modules/programs/eww.nix

```diff
@@ -23,12 +21,74 @@ in {
       '';
     };
 
-    configDir = mkOption {
-      type = types.path;
-      example = literalExpression "./eww-config-dir";
+    yuck = {
+      enable = mkEnableOption ""
+      mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = ''
+          (defwindow example
+            :monitor 0
+            :geometry (geometry :x "0%"
+                                 :y "20px"
+                                 :width "90%"
+                                 :height "30px"
+                                 :anchor "top center")
+            :stacking "fg"
+            :reserve (struts :distance "40px" :side "top")
+            :windowtype "dock"
+            :wm-ignore false
+          "example content")
+        '';
+        description = ''
+          The content that gets symlinked to
+          {file} `$XDG_CONFIG_HOME/eww/eww.yuck`.
+        '';
+      };
+    };
+
+    scss = {
+      enable = mkEnableOption "Enables string based scss config";
+      config = mkOption {
```

Comment:
I would suggest `extraConfig` as name instead. The problem with `config` is that it is a very overloaded term and the convention is to use the `extraConfig` name (even though there are no other options).

Reply:
Actually, looking at it closer I agree with @mainrs, just make single `scssConfig` and `yuckConfig` options of type `lines`, default it to `""` and skip creating the file if the string is empty.

Reply:
I was recommend by someone named `just_midi` in discord to do this as they said they were maintainer of multiple modules in home manager, can't find their gh username unfortantly

Reply:
> Actually, looking at it closer I agree with @mainrs, just make single `scssConfig` and `yuckConfig` options of type `lines`, default it to `""` and skip creating the file if the string is empty.

wouldn't null just be better then? if the type is not null then create the file else not.